File: dongyang_algo/day04/ZeroFinderStr.java
package day04;

public class ZeroFinderStr {
    
    public static void main(String[] args) {
        int[] arr={6,0,8,2,3,0,4,0,1};
        
        int j=0;
        for(int i=0; i<arr.length; i++){
            if(arr[i]!=0){
                arr[j]=arr[i];
                j++;
            }
        }
        for(int i=j; i<arr.length;i++){
            arr[i]=0;
        }

        for(var i: arr){
            System.out.printf("%d ", i);
        }
    }
}

File: dongyang_algo/day04/ZeroFinderStr.py
def reorder(arr, n):
    j = 0  # Count of non-zero elements
    for i in range(n):
        if arr[i] != 0:
            arr[j] = arr[i]
            j += 1

    for i in range(k, n):
        ar[i] = 0


# code
arr = [6, 0, 8, 2, 3, 0, 4, 0, 1]
n = len(arr)
reorder(arr, n)
print(arr)
